src/vs/platform/remote/browser/browserSocketFactory.ts

```diff
@@ -208,7 +208,12 @@ export class BrowserSocketFactory implements ISocketFactory {
 	}
 
 	connect(host: string, port: number, query: string, callback: IConnectCallback): void {
-		const socket = this._webSocketFactory.create(`ws://${host}:${port}/?${query}&skipWebSocketFrames=false`);
+		let wsProtocol = 'ws';
+		if (location.protocol === 'https:') {
+			wsProtocol = 'wss';
+		}
+
+		const socket = this._webSocketFactory.create(`${wsProtocol}://${host}:${port}/?${query}&skipWebSocketFrames=false`);
```

Comment:
needs sql carbon edit, also can you just make this one line? useful for smaller merge conflicts 

```` typescript
const socket = this._webSocketFactory.create(`${location.protocol === 'https:' ? wss : ws}://${host}:${port}/?${query}&skipWebSocketFrames=false`);
````